apps/web/src/components/Basenames/UsernameProfileContext/index.tsx

```diff
@@ -1,61 +1,126 @@
 'use client';
 import { BaseName } from '@coinbase/onchainkit/identity';
+import { USERNAME_L2_RESOLVER_ADDRESSES } from 'apps/web/src/addresses/usernames';
+import useBaseEnsName from 'apps/web/src/hooks/useBaseEnsName';
+import useBasenameChain from 'apps/web/src/hooks/useBasenameChain';
+import { buildBasenameOwnerContract } from 'apps/web/src/utils/usernames';
 import {
   Dispatch,
   ReactNode,
   SetStateAction,
   createContext,
+  useCallback,
   useContext,
   useMemo,
   useState,
 } from 'react';
 import { Address } from 'viem';
-import { useAccount } from 'wagmi';
+import { useAccount, useEnsAddress, useReadContract } from 'wagmi';
 
 export enum UsernameProfileSteps {}
 
 export type UsernameProfileContextProps = {
+  // Profile details
   profileUsername: BaseName;
   profileAddress?: Address;
-  currentWalletIsOwner?: boolean;
+
+  // Profile owner
+  profileOwnerAddress?: Address;
+  profileOwnerUsername?: BaseName;
+  profileOwnerRefetch: () => Promise<void>;
+
+  // State
+  currentWalletIsProfileOwner: boolean;
+  currentWalletIsProfileAddress: boolean;
+
+  // Settings
   showProfileSettings: boolean;
   setShowProfileSettings: Dispatch<SetStateAction<boolean>>;
 };
 
 export const UsernameProfileContext = createContext<UsernameProfileContextProps>({
   profileUsername: 'default.basetest.eth',
   profileAddress: undefined,
-  currentWalletIsOwner: false,
+  profileOwnerAddress: undefined,
+  profileOwnerRefetch: async () => undefined,
+  currentWalletIsProfileOwner: false,
+  currentWalletIsProfileAddress: false,
   showProfileSettings: false,
   setShowProfileSettings: () => undefined,
 });
 
 type UsernameProfileProviderProps = {
   children: ReactNode;
   username: BaseName;
-  address: Address;
 };
 
 export default function UsernameProfileProvider({
   children,
   username,
-  address,
 }: UsernameProfileProviderProps) {
-  const profileUsername = username;
-
   const [showProfileSettings, setShowProfileSettings] = useState<boolean>(false);
-  const { address: connectedAddress } = useAccount();
-  const currentWalletIsOwner = connectedAddress === address;
+  const { basenameChain } = useBasenameChain(username);
+
+  // Current wallet
+  const { address: connectedAddress, isConnected } = useAccount();
+
+  // Profile address
+  const {
+    data: profileAddress,
+    isFetching: profileAddressIsFetching,
+    refetch: profileAddressRefetch,
+  } = useEnsAddress({
+    name: username,
+    universalResolverAddress: USERNAME_L2_RESOLVER_ADDRESSES[basenameChain.id],
+    query: {
+      retry: false,
+    },
```

Comment:
why was retry specifically disabled here?

Reply:
Most of the time avatar/name are null, by default react-query will retry 3 times when getting no data.

But in this case no data is expected.

Reply:
nice